File: 1442-CountTripletsThatCanFormTwoArraysOfXorEqual.java
// https://leetcode.com/problems/count-triplets-that-can-form-two-arrays-of-equal-xor/
class Solution {
    public int countTriplets(int[] arr) {
        int length = arr.length;
        int count = 0,a = 0, b = 0;
        for (int i = 0; i < length - 1; i++) {
            a = 0;
            for (int j = i + 1; j < length; j++) {
                a ^= arr[j - 1];
                b = 0;
                for (int k = j; k < length; k++) {
                    b ^= arr[k];
                    if (a == b) {
                        count += 1;
                    }
                }
            }
        }
        return count;
    }
}

File: 1442-count_triplets_that_can_form_two_arrays_of_xor_equal.py
# https://leetcode.com/problems/count-triplets-that-can-form-two-arrays-of-equal-xor/
class Solution:
    def countTriplets(self, arr: List[int]) -> int:
        count = 0
        length = len(arr)
        for i in range(length - 1):
            a = 0
            for j in range(i+ 1, length):
                a ^= arr[j - 1]
                b = 0
                for k in range(j, length):
                    b ^= arr[k]
                    if (a == b):
                        count += 1
        return count
                
